core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -346,6 +346,25 @@ final class Stream[+F[_], +O] private[fs2] (private[fs2] val underlying: Pull[F,
     go(Nil, false, this).stream
   }
 
+  // A builder for queues.
+  trait MakeQueue[F[_]] {
```

Comment:
I'm not really sure where to put this, but it lets us create a queue-variant chosen by the user but they can't tamper with its contents at all since `create` is polymorphic.